### PR TITLE
Adopt code for Khalani Protocol

### PR DESCRIPTION
**Pull request by serejke**
This PR contains the patches on top of the origin Avalanche Faucet that adopt for Khalani protocol:
- allow to configure faucet from a file specified by `CONFIG_FILE`
- allow to override RPC URLs with ENV variables of name `EVM_CHAINS_${ID}_RPC`, where `ID` is the chain ID specified in `evmchains` of `config.json`, for example `EVM_CHAINS_KHALANITESTNET_RPC`
- allow to disable captcha verification — needed to test the deployment and then register Google Captcha for the production domain
- added Khalani Protocol branding (logo / docs)
- GitHub pipeline to build a Docker image

Closes https://github.com/tvl-labs/infrastructure/issues/213
Closes https://github.com/tvl-labs/infrastructure/issues/214

### CODE REVIEW

File: .env.dist [hunk 1]
```diff
@@ -1,3 +1,5 @@
-CAPTCHA_SECRET="Google ReCaptcha V3 Secret"
-V2_CAPTCHA_SECRET="Google ReCaptcha V2 Secret"
+EVM_CHAINS_KHALANITESTNET_RPC=https://www.axon-node.info
+DISABLE_CAPTCHA=true
```
Comment (distributedstatemachine) [inline .env.dist:7]: Why are you disabling captcha ? Faucets get DDOSed (tbh its a nice problem for us to have at this point ) . I would like to understand the intuition behind this

File: client/public/index.html [hunk 2]
```diff
@@ -2,21 +2,22 @@
...
 
-    <meta name="description" content="Get testnet tokens to use on the Avalanche Fuji Network (testnet) or a Subnet (WAGMI, Defi Kingdoms, Dexalot, Swimmer, and Castle Crush) in one place.">
-    <meta name="keywords" content="Avalanche, AVAX, Faucet, Fuji, Testnet">
-    
-    <meta property="og:title" content="Avalanche Faucet">
-    <meta property="og:site_name" content="Faucet | Avalanche">
-    <meta property="og:url" content="https://faucet.avax.network/">
-    <meta property="og:description" content="Get testnet tokens to use on the Avalanche Fuji Network (testnet) or a Subnet (WAGMI, Defi Kingdoms, Dexalot, Swimmer, and Castle Crush) in one place.">
+
+    <meta name="description" content="Get testnet tokens to use for the Khalani Protocol (USDC/USDT testnet tokens on Khalani chain, Goerli, Avalanche Fuji, etc) in one place.">
```
Comment (distributedstatemachine) [inline client/public/index.html]: Do you want to take out chain specific details  `USDC/USDT testnet tokens on Khalani chain, Goerli, Avalanche Fuji, etc` and replace it with something along the lines of "Testnet tokens to interact with the Khalani protocol"

File: client/public/index.html [hunk 3]
```diff
@@ -2,21 +2,22 @@
...
+    <meta name="keywords" content="Khalani, Faucet">
+
+    <meta property="og:title" content="Khalani Faucet">
+    <meta property="og:site_name" content="Faucet | Khalani">
+    <!-- TODO: set the site URL -->
+    <meta property="og:url" content="https://khalani-faucet.digipnyx.org/">
+    <meta property="og:description" content="Get testnet tokens to use for the Khalani Protocol (USDC/USDT testnet tokens on Khalani chain, Goerli, Avalanche Fuji, etc) in one place.">
```
Comment (distributedstatemachine) [inline client/public/index.html]: Same as above

**Review by e00dan — APPROVED** [review]

**Review by distributedstatemachine — CHANGES_REQUESTED** [review]
This need to be modified to `Add chain to Metamask`, as subnets are an avalanche specific thing

File: .env.dist (hunk 1 above)
Comment (distributedstatemachine) [reply .env.dist:7]: seen this : 

> - allow to disable captcha verification — needed to test the deployment and then register Google Captcha for the production domain

**Comment by serejke** [comment]
A lot of code has been re-implemented on the server side:
- properly handle errors
- significantly streamlined the state management of requests and nonce/balance recalibration (`mem-pool` -> `signed` -> `sent` OR `error`)
- fixed a critical blocker that the code used to sign the transaction twice
- added support of faucet wallet signer with [AWS KMS](https://docs.aws.amazon.com/kms/latest/APIReference/API_Sign.html)
- added Prometheus support 

Next steps:
- deploy this PR with full KMS and Prometheus EKS setup
- see if captcha / rate limit changes are necessary, if not then it can be merged

File: server.ts [hunk 4]
```diff
@@ -1,184 +1,225 @@
...
-    const evm: EVMInstanceAndConfig = evms.get(chain)!
+    // GET request for fetching all the chain and token configurations
+    router.get('/getChainConfigs', (req: any, res: any) => {
+        const configs: any = [...configFile.evmchains, ...configFile.erc20tokens]
+        res.send({ configs })
+    })
 
-    let balance: BN = evm?.instance.getBalance(erc20)
+    // GET request for fetching faucet address for the specified chain
+    router.get('/faucetAddress', async (req: any, res: any) => {
```
Comment (konraddb) [inline server.ts]: no need 'async' here

File: server.ts [hunk 5]
```diff
@@ -1,184 +1,225 @@
...
+        const id: string | undefined = req.query?.erc20
+        const evm = evmMap.get(chain);
+        if (!evm) {
+            res.status(400).send({ message: `No chain found ${chain}!` })
+            return;
+        }
 
-    res.status(200).send({
-        balance: balance?.toString()
+        const balance = evm.getBalance(id);
```
Comment (konraddb) [inline server.ts:178]: In the source code, when there is no balance, their are using `balance = new BN(0)`

File: client/public/banner.jpeg
Comment (konraddb) [inline client/public/banner.jpeg:1]: Can we perhaps use a logo here with a transparent background and of better quality?

**Review by konraddb — COMMENTED** [review]
I compared the source code with the refactored code and did not find any potential issues. However, by the way, a few cosmetic issues have been reported.

**Comment by distributedstatemachine** [comment]
@serejke is this good to go?

**Comment by serejke** [comment]
Merged 🚀

**Pull request merged**